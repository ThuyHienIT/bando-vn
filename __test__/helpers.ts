import { randomUUID } from 'crypto';

export function generateFacility(
  type: FacilityTypeEnum,
  name: string = 'Venus Meeting Room'
): FacilityItem {
  return {
    phoneNumber: '24709620',
    id: randomUUID(),
    address: '1 JALAN KILANG,S159402',
    description:
      'Current majority suffer hair born.\nReport daughter government item keep political cut. Customer apply couple. Data happen hear must almost.',
    name,
    amenities:
      'Friend ok nation international line another within.\nUp rate when sometimes probably measure with. Product town sort. Include even alone certain.',
    type: type,
    operationHours: ['8:00', '19:00'],
    offDays: ['2022-04-23T23:18:53Z'],
    thumbnail: 'https://picsum.photos/400/200',
  };
}

export function generateBooking(facilityId: string): BookingItem {
  return {
    facilityId: facilityId,
    from: '2023-04-23T10:30:00Z',
    to: '2023-04-23T11:00:00Z',
    userEmail: 'kqthang1505@gmail.com',
    id: randomUUID(),
    createdAt: '2023-02-18T22:25:56+08:00',
    updatedAt: '2023-02-18T22:25:56+08:00',
  };
}
