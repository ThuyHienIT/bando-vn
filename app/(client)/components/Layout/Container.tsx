'use client';

import { Layout } from 'antd';
import { memo } from 'react';
import styled from 'styled-components';

const ContainerStyle = styled.div`
  max-width: 1080px;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
`;

interface Props {
  children: React.ReactNode;
  className?: string;
}
export const Container = memo<Props>(function Header(props) {
  return <ContainerStyle className={props.className}>{props.children}</ContainerStyle>;
});
