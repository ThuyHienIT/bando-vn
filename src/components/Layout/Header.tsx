import { Layout, Typography } from 'antd';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { memo, useCallback } from 'react';
import styled from 'styled-components';

const HeaderStyle = styled(Layout.Header)`
  position: sticky;
  top: 0;
  z-index: 1;
  width: 100%;
  background-color: #f0f2f5;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid #dddddd;
`;

const LogoStyle = styled(Link)`
  height: 31px;
  margin: 16px 24px 16px 0;
  background: rgba(255, 255, 255, 0.2);
  display: inline-flex;
  align-items: center;
  gap: 8px;
`;

const Nav = styled.div`
  flex-grow: 1;
`;

interface Props {}

export const Header = memo<Props>(function Header(props) {
  const router = useRouter();
  console.log('router', router);

  const handleClick = useCallback(
    ({ key }: { key: string }) => {
      router.push(`/admin/${key}`);
    },
    [router]
  );

  return (
    <HeaderStyle>
      <LogoStyle href="/admin">
        <img src="/logo.png" alt="Facility Booking" height={30} />
        <Typography.Title level={4} style={{ margin: 0 }}>
          VN Map - Admin
        </Typography.Title>
      </LogoStyle>
    </HeaderStyle>
  );
});
