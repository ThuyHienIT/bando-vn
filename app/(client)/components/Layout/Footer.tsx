'use client';

import { Layout, Typography } from 'antd';
import { memo } from 'react';
import styled from 'styled-components';
import { Container } from './Container';

interface Props {}
export const Footer = memo<Props>(function Header(props) {
  return (
    <Layout.Footer style={{ padding: '8px 0' }}>
      <Container>
        <Typography.Text type="secondary">
          <small>Â© Created by Thang Kieu</small>
        </Typography.Text>
      </Container>
    </Layout.Footer>
  );
});
