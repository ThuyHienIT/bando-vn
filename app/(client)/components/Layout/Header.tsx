'use client';

import { ProfileOutlined } from '@ant-design/icons';
import { Layout, Typography } from 'antd';
import Image from 'next/image';
import { memo } from 'react';
import styled from 'styled-components';
import { UserProfile } from '../UserProfile';
import { Container } from './Container';

const HeaderStyle = styled(Layout.Header)`
  &&& {
    background-color: #fff;
  }
`;

const HeaderContent = styled(Container)`
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const Logo = styled.div`
  display: flex;
  align-items: center;
  gap: 8px;

  .ant-typography {
    margin-bottom: 0;
  }
`;

interface Props {}
export const Header = memo<Props>(function Header(props) {
  return (
    <HeaderStyle>
      <HeaderContent>
        <Logo>
          <Image src="/logo.png" alt="Facility Booking" width={30} height={30} />
          <Typography.Title level={4}>Facility Booking</Typography.Title>
        </Logo>
        <UserProfile />
      </HeaderContent>
    </HeaderStyle>
  );
});
