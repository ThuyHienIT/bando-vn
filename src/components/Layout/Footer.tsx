import { Layout, Typography } from 'antd';
import { memo } from 'react';
import styled from 'styled-components';

import { Container } from './Container';

const FooterStyle = styled(Layout.Footer)`
  padding: 8px 0;
  border-top: 2px solid #dddddd;
`;

interface Props {}
export const Footer = memo<Props>(function Header(props) {
  return (
    <FooterStyle style={{ padding: '8px 0' }}>
      <Container>
        <Typography.Text type="secondary">
          <small>Â© Created by Thang Kieu</small>
        </Typography.Text>
      </Container>
    </FooterStyle>
  );
});
