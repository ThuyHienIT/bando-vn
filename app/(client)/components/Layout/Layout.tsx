'use client';

import { Layout } from 'antd';
import { memo } from 'react';
import { Header } from './Header';
import styled from 'styled-components';
import { Footer } from './Footer';
import { Container } from './Container';

const Content = styled(Layout.Content)``;

interface Props {
  children: React.ReactNode;
}
export const BasicLayout = memo<Props>(function BasicLayout(props) {
  return (
    <Layout>
      <Header />
      <Content className="site-layout">
        <Container>{props.children}</Container>
      </Content>
      <Footer />
    </Layout>
  );
});
