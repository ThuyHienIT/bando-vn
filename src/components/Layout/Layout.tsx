import { Layout } from 'antd';
import { memo, useEffect } from 'react';
import { useRecoilState } from 'recoil';
import styled from 'styled-components';

import { userInfoState } from '@recoil/user';

import { Container } from './Container';
import { Footer } from './Footer';
import { Header } from './Header';

const LayoutStyle = styled(Layout)`
  &&& {
    min-height: 100vh;
  }
`;

interface Props {
  children: React.ReactNode;
  userInfo?: UserInfo;
}
export const BasicLayout = memo<Props>(function BasicLayout(props) {
  const [userInfo, updateUserInfo] = useRecoilState(userInfoState);

  useEffect(() => {
    if (!userInfo && props.userInfo) {
      updateUserInfo(props.userInfo);
    }
  }, [props.userInfo]);

  return (
    <LayoutStyle>
      <Header />
      <Layout.Content>
        <Container className="main-container">{props.children}</Container>
      </Layout.Content>
      <Footer />
    </LayoutStyle>
  );
});
