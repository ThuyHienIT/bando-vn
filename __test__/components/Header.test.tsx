// import Home from '../pages/index'
import '@testing-library/jest-dom';

import { userInitialState } from '__test__/helpers';
import { RecoilRoot } from 'recoil';
import { Header } from 'src/(client)/components/Layout/Header';

import { render, screen } from '@testing-library/react';

describe('Header', () => {
  it('renders a heading', () => {
    render(
      <RecoilRoot initializeState={userInitialState}>
        <Header />
      </RecoilRoot>
    );

    const heading = screen.getByRole('heading', {
      name: /facility booking/i,
    });

    expect(heading).toBeInTheDocument();
  });

  it('renders header unchanged', () => {
    const { container } = render(
      <RecoilRoot initializeState={userInitialState}>
        <Header />
      </RecoilRoot>
    );
    expect(container).toMatchSnapshot();
  });
});
