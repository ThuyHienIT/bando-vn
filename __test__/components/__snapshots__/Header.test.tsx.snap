// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders header unchanged 1`] = `
.c1 {
  max-width: 1080px;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
}

.c0 {
  border-bottom: 2px solid #dddddd;
}

.c0.c0.c0 {
  background-color: #fff;
  padding-inline: 0;
}

.c2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.c3 .ant-typography {
  margin-bottom: 0;
}

<div>
  <header
    class="ant-layout-header c0"
  >
    <div
      class="c1 c2"
    >
      <a
        href="/"
      >
        <div
          class="c3"
        >
          <img
            alt="Facility Booking"
            data-nimg="1"
            decoding="async"
            height="30"
            loading="lazy"
            src="/_next/image?url=%2Flogo.png&w=64&q=75"
            srcset="/_next/image?url=%2Flogo.png&w=32&q=75 1x, /_next/image?url=%2Flogo.png&w=64&q=75 2x"
            style="color: transparent;"
            width="30"
          />
          <h4
            class="ant-typography css-dev-only-do-not-override-diro6f"
          >
            Facility Booking
          </h4>
        </div>
      </a>
      <button
        class="ant-btn css-dev-only-do-not-override-diro6f ant-btn-circle ant-btn-default ant-dropdown-trigger"
        type="button"
      >
        <span
          aria-label="user"
          class="anticon anticon-user"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="user"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </span>
      </button>
    </div>
  </header>
</div>
`;
